source/mongodb-handshake/handshake.rst

```diff
@@ -418,11 +474,78 @@ the ``platform`` field before all others. Additionally, implementors are
 encouraged to place high priority information about the platform earlier in the
 string, in order to avoid possible truncating of those details.
 
+If the ``client.env`` field would be populated but would cause the metadata document
+to exceed the size limit, drivers MUST clear all fields of ``client.env`` except for
+``client.env.name``, and if that would still exceed the size limit drivers MUST omit
+``client.env`` entirely.
```

Comment:
Regarding the last part:
> if that would still exceed the size limit drivers MUST omit `client.env` entirely.

Reading this along with the paragraph before I'm not entirely certain whether drivers should first truncate the `platform` field or omit `env`. This could be clarified here, as I believe it's better to truncate `platform` before removing `client.env.name`.
This would give us the following priority list (in descending order of importance):
* `application.name`
* `driver.*`
* `os.type`
* `env.name`
* `os.*` (except `type`)
* `env.*` (except `name`)
* `platform`

If the handshake metadata size exceeds 512 bytes, drivers can then use this list to decide what information to truncate to get to 512 bytes.

Reply:
Ah, yup.  Updated the section to lay this out.